File: app/src/main/java/com/example/sampleproject_1/WaterReminder/Database/DatabaseWaterReminder.java
package com.example.sampleproject_1.WaterReminder.Database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {EntityWaterReminder.class}, version = 4)
public abstract class DatabaseWaterReminder extends RoomDatabase {

    private static final String dbName = "database_water_reminder";

    private static DatabaseWaterReminder instance;

    public abstract DataAccessObjectWaterReminder dataAccessObjectWaterReminder();

    public static synchronized DatabaseWaterReminder getInstance(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(), DatabaseWaterReminder.class, dbName)
                    .fallbackToDestructiveMigration()
                    .build();
        }
        return instance;
    }
}

File: app/src/main/java/com/example/sampleproject_1/WaterReminder/Database/DatabaseWaterReminder.kt
package com.example.sampleproject_1.WaterReminder.Database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [EntityWaterReminder::class], version = 7)
abstract class DatabaseWaterReminder : RoomDatabase() {
    abstract fun dataAccessObjectWaterReminder(): DataAccessObjectWaterReminder

    companion object {
        private const val dbName = "database_water_reminder"

        private var instance: DatabaseWaterReminder? = null

        @JvmStatic
        @Synchronized
        fun getInstance(context: Context): DatabaseWaterReminder? {
            if (instance == null) {
                instance = Room.databaseBuilder(context.applicationContext, DatabaseWaterReminder::class.java, dbName)
                        .fallbackToDestructiveMigration()
                        .build()
            }

            return instance
        }
    }
}